src/api/fetchData.js

```diff
@@ -2,24 +2,29 @@ import { useState, useEffect } from 'react';
 
 export default function fetchData(url) {
     const [data, setData] = useState([]);
-    const [loading, setLoading] = useState(null);
+    const [loading, setLoading] = useState(true);
     const [error, setError] = useState('');
 
+    const rawgioApiKey = '';
```

Comment:
you can use  import.meta.env.API_KEY to bring the key in a safe environment.

You should have a .env.development.local with:
API_KEY='your API key string'
REACT_APP_API_URL = "http://localhost:3000"